Config split 2

pyrello opened the pull request:
Related to #4035
Resolves #4137 
Resolves #1626

# How to test
* Add your name after the split so others know you are testing it.
* `git checkout config_split_2 && git pull && composer install`
* Test feature splits:
  * Review file changes ([instructions](https://github.com/uiowa/uiowa/pull/4091#issuecomment-956973725))
  * `drush @default.local sql:drop && blt ds && drush @default.local uli admin/config/development/configuration/config-split`
  * Enable the feature split you are testing
  * `drush @default.local cim`
  * Test for regressions
  - [x] additional_person_types - @pyrello
  - [x] apr - @pyrello
  - [x] areas_of_study - @e-marie-w 
  - [x] event - @cory-skeers 
  - [x] grad_person_extended - @cory-skeers 
  - [x] profiles - @pyrello 
  - [x] scholarship - @cory-skeers 
  - [x] sitenow_v2 (@joewhitsitt)
  - [x] thesis_defense - @e-marie-w 
  - [x] topic_page - @quamsta 
  - [x] uiowa_graph - @pyrello 
* For each site split:
  * Review file changes ([instructions](https://github.com/uiowa/uiowa/pull/4091#issuecomment-956973725))
  * `drush @site.local sql:drop && blt ds --site site && drush @site.local uli`
  * Test for regressions
  - [x] admissions.uiowa.edu - _multi-split_
  - [x] amcs.uiowa.edu - _multi-split_ - @pyrello 
  - [x] brand.uiowa.edu (@joewhitsitt)
  - [x] coronavirus.uiowa.edu (@richardbporter)
  - [x] ~~education.uiowa.edu~~  - _empty, deleted_ - (@richardbporter)
  - [x] genetics.grad.uiowa.edu - @e-marie-w 
  - [x] grad.uiowa.edu - _multi-split_ - @pyrello 
  - [x] hr.uiowa.edu (@joewhitsitt)
  - [x] icsa.uiowa.edu (@joewhitsitt)
  - [x] ighn.international.uiowa.edu (@joewhitsitt)
  - [x] immuno.grad.uiowa.edu - _multi-split_ - @pyrello
  - [x] iowasuperfund.uiowa.edu (@richardbporter)
  - [x] pharmacy.uiowa.edu (@richardbporter)
  - [x] policy.clas.uiowa.edu (@joewhitsitt)
  - [x] presidentialsearch.uiowa.edu (@richardbporter)
  - [x] sitenow.uiowa.edu (@richardbporter)
  - [x] sppa.uiowa.edu (@richardbporter)
  - [x] tippie.uiowa.edu - @makurj 
  - [x] transportation.uiowa.edu (@richardbporter)
  - [x] uiowa.edu (@joewhitsitt)

pyrello committed:
Switch to config_split 2.x branch.

pyrello committed:
Updated config_split configs to new format.

pyrello committed:
Updated partial split config for topic_page.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Re-exported configuration usingcode updates from https://git.drupalcode.org/project/config_split/-/merge_requests/9/diffs.

pyrello committed:
Merged 'main' branch, fixed conflicts, updated composer.lock.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Adding patch to with necessary updates for CS2.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Re-exported to match expected order after fresh install.

pyrello committed:
Re-exported topic split items with new format.

pyrello committed:
Merge branch 'main' into config_split_2

bspeare committed:
Merge branch 'main' into config_split_2

pyrello committed:
Re-exported apr split to update permissions.

bspeare committed:
Exported areas_of_study

bspeare committed:
Added deleted files for areas_of_study

bspeare committed:
Exported scholarship feature

pyrello committed:
Updated event split.

pyrello committed:
Merged with 'main' branch, fixed conflicts, updated composer.lock.

pyrello committed:
Added editorial workflow partial.

pyrello committed:
Fix webmaster permissions for scholarship feature.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Re-exported events. A little bit of weirdness here that we'll have to come back to.

bspeare committed:
Exported thesis_defense

bspeare committed:
Exported profiles

bspeare committed:
Exported grad_person_extended

pyrello committed:
Merged 'main' branch, resolved conflicts, and re-generated composer.lock.

pyrello committed:
Moving to a different patch to solve one problem at a time. Re-exported grad_person_extended.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Add patch andd remove non-patch files that should be patched.

pyrello committed:
Merged 'main' branch, resolved conflicts, and updated composer.lock.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Added another patch to the config_split module to solve a problem where multiple versions of a split configuration were being exported. Updated the grad_person_extended split and re-exported it.

pyrello committed:
Updated uiowa_graph split to new format.

pyrello committed:
Merged 'main' branch, fixed conflicts, and updated composer.lock.

pyrello committed:
Re-exported apr split.

pyrello committed:
Re-exported areas_of_study split.

pyrello committed:
Needed to export this differently.

pyrello committed:
Adding some of the re-exported config from event split. Still need to investigate some of it.

pyrello committed:
Updated event split definition.

joewhitsitt committed:
re-export topic_page

joewhitsitt committed:
move topic collection sitemap to complete

joewhitsitt committed:
re-export, reorganize thesis_defense

pyrello committed:
Merged 'main' branch, fixed conflicts (although need to re-export some user.role.* configs, and updated composer.lock.

pyrello committed:
Updated to most recent version of CS2.

pyrello committed:
Added patch with new test to make testing easier.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Removed patch that is no longer necessary.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Merged 'main' branch, resolved conflicts, and updated composer.lock.

pyrello committed:
Fixed export for event using new fix.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Cleaning up export of areas_of_study split.

pyrello committed:
Updated topic_page split and re-exported.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Updated to using actual static patch files for config_split patches and pinned the dev version of config_split for now to ensure stability moving forward.

pyrello committed:
Re-exported the scholarship feature.

pyrello committed:
A couple manual fixes to account for inconsistencies.

pyrello committed:
Updated scholarship split.

pyrello committed:
Re-exported updated event split.

pyrello committed:
Re-exported profiles split.

pyrello committed:
Re-exported thesis_defense split. Manually cleaned up permissions that weren't supposed to be there.

pyrello committed:
Added patch (temporarily until it gets merged) to change the keys used in split patches to make them more readable.

pyrello committed:
Re-exported areas of study split after patch to make keys more readable in patch files.

pyrello committed:
Merged main branch, fixed conflicts, updated composer.lock, and re-exported grad_person_extended.

pyrello committed:
Re-exported grad_person_extended.

pyrello committed:
Merged main branch, fixed conflicts, updated composer.lock.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Switched to new patch.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
An attempt at re-exporting the admissions site split.

pyrello committed:
Fix test to not use mixed array type.

pyrello committed:
Trying out patch for sequential split handling.

pyrello committed:
Updated commit hash for config split and removed two patches that are no longer necessary.

pyrello committed:
Trying out adding mode = linked to feature splits.

pyrello committed:
Didn't successfully update the commit hash for config_split last time, so needed another commit to do it.

pyrello committed:
Updated patch.

pyrello committed:
Re-exported event feature split, which updated patch exports.

pyrello committed:
Re-exported event feature split, which updated patch exports.

pyrello committed:
Updated commit hash for config_split. Trying out a new patch.

pyrello committed:
Update to admissions config item.

pyrello committed:
Re-exported the grad.uiowa.edu site split.

pyrello committed:
Added some grad site config that was missed in previous commit.

pyrello committed:
Updated to new patch for resolving site split issues.

pyrello committed:
Re-exported sitenow_v2 config split.

pyrello committed:
Merged main branch and fixed conflicts.

pyrello committed:
Merged 'main' branch, fixed conflicts, and updated composer.lock.

pyrello committed:
Re-exported split definitions.

pyrello committed:
Re-exported brand site split.

pyrello committed:
Adding files I missed in the last commit.

pyrello committed:
Re-exported coronavirus site split.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Re-exported admissions and grad splits.

pyrello committed:
Switch feature splits to being stackable.

pyrello committed:
Re-exported the grad immuno site split. Getting both a patch and complete split file for views.view.person_custom still :|

pyrello committed:
Re-exported the HR site split. A few items to investigate in this batch.

pyrello committed:
Re-exported ICSA.

pyrello committed:
Re-exported transportation site split.

pyrello committed:
Re-exported tippie site split.

pyrello committed:
Re-exported sppa site split.

pyrello committed:
Re-exported sitenow site split.

pyrello committed:
Re-exported pharmacy site split.

pyrello committed:
Re-exported presidential search site split. Added some files from a previous export that I missed.

pyrello committed:
Re-exported iowasuperfund site split.

pyrello committed:
Re-exported ighn international site split.

pyrello committed:
Re-exported clas policy site split.

pyrello committed:
Re-exported the homepage site.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Re-exported APR feature split. Added some files for uiowa.edu split that I missed previously.

pyrello committed:
Added readme with APR stories.

pyrello committed:
Updated the APR readme.

pyrello committed:
Added README for additional_person_types split. Modified name to APR README.

pyrello committed:
Modified name to APR README again.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Adjusted the weight of additional_person_types split to match other feature splits.

pyrello committed:
Re-exported genetics site split.

pyrello committed:
Added README file to uiowa_graph split.

pyrello committed:
Added README to profiles split.

pyrello committed:
Updated grad split and re-exported.

pyrello committed:
Added beginning of README for grad split.

pyrello committed:
Merge branch 'main' into config_split_2

Commit:
Update sitenow role mappings with new AD groups

richardbporter commented on config/sites/sitenow.uiowa.edu/config_split.patch.uiowa_auth.settings.yml:
```diff
@@ -1,15 +1,9 @@
     - 'requester|urn:oid:1.3.6.1.4.1.5923.1.1.1.5|staff'
     - 'requester|urn:oid:1.3.6.1.4.1.5923.1.1.1.5|affiliate'
     - 'requester|urn:oid:1.3.6.1.4.1.5923.1.1.1.5|member'
   all_valid_hawkids_can_authenticate: true
```
I've no idea what this `all_valid_hawkids_can_authenticate` is. Its in main but I don't think it should be. Going to remove...

joewhitsitt replied:
i plan to do the same for brand

Commit:
Remove rogue auth config item

pyrello committed:
Removed permissions that shouldn't be there.

cory-skeers commented on config/features/grad_person_extended/config_split.patch.core.entity_form_display.node.person.default.yml:
```diff
@@ -0,0 +1,177 @@
+    field_person_research_areas:
+      weight: 17
+      settings:
+        match_operator: STARTS_WITH
```
It's in the original `grad_person_extended` split and not introduced by config split 2 work, but wondering if this was intended to change from `CONTAINS` to `STARTS_WITH`? If not, is it easier to change it here, or separately and re-export @pyrello?

pyrello replied:
I'm guessing it got changed at some point between when this field started out in this split and then moved into the default config. I'd opt for as few differences as possible. Not sure about the best way to go about changing it off the top of my head. Possibly change it in the config file, run `drush @default.local cim`, then re-export? I'm assuming it would exclude it then.

pyrello committed:
Updated thesis_defense split and re-exported.

pyrello committed:
Re-exported topic page split.

pyrello committed:
Re-exported scholarship split.

pyrello committed:
Re-exported profiles split.

cory-skeers committed:
Update grad_person_extended split to not diff match_operator.

joewhitsitt committed:
add readme for custom v2 sites.

joewhitsitt committed:
Merge branch 'config_split_2' of github.com:uiowa/uiowa into config_split_2

pyrello committed:
Re-re-exported sitenow_v2.

pyrello committed:
Updated sitenow_v2 split and added a file that got missed.

pyrello committed:
Two additional config files that are necessary.

pyrello committed:
Updated sitenow_v2 split.

pyrello committed:
Manually updating permission changes that I don't believe were intentional.

pyrello committed:
Removing permissions that seem to be unintentional in the admissions site split.

richardbporter commented on config/sites/presidentialsearch.uiowa.edu/config_split.patch.user.role.webmaster.yml:
```diff
@@ -0,0 +1,23 @@
+adding: {  }
+removing:
+  permissions:
+    - 'access any webform configuration'
```
Are these permission removals intentional? I think webform might be but can't quite remember. Panopto perms were maybe just never updated.

richardbporter replied:
Removed panopto perms.

richardbporter replied:
Seems like it: https://github.com/uiowa/uiowa/pull/2679

Commit:
Add sitenow.uiowa.edu split readadme

Commit:
Grant webmaster panopto perms in presidentialsearch.uiowa.edu split

Commit:
Add presidentialsearch.uiowa.eduu split readme

pyrello committed:
Updated config_split to most recent commit.

Commit:
Add coronavirus.uiowa.edu site split readme

Commit:
Add education split readme

Commit:
Export amcs.uiowa.edu split after config_split 8002 update

Commit:
Add amcs.uiowa.edu split readme

Commit:
Export pharmacy_migrate split after config_split 8002 update

Commit:
Make icsa and sitenow splits stackable

Commit:
Export transportation_migrate split after config_split 8002 update

Commit:
Remove edit panopto media perm split removal from transportation

joewhitsitt committed:
fix check for v2 split. plugin.manager.config_filter can't be relied on anymore

joewhitsitt committed:
add v2 readme

pyrello committed:
Updates to immuno grad site split.

pyrello committed:
Merge branch 'main' into config_split_2

Commit:
Manually update transportation split core.entity_view_display.node.page.default to add Transportation block category

richardbporter committed:
Merge branch 'main' into config_split_2

pyrello committed:
Updated to more recent dev commit hash for config_split. Re-exported immuno grad site split after updates to patch.

pyrello committed:
Switching page view display config to use complete for now.

joewhitsitt committed:
more hopeful fixes for sitenow_v2

joewhitsitt committed:
undo these. permissions moot. can't get to this point to interact with these blocks

pyrello committed:
Merge branch 'main' into config_split_2

joewhitsitt committed:
attempt re-export, clean up of HR

joewhitsitt committed:
more patch, less complete for HR

pyrello committed:
Honestly not sure what fixed this.

pyrello committed:
Updated config_split to most recent changes to module and patch.

pyrello committed:
Merged in 'main' branch, fixed conflicts, and updated composer.lock.

pyrello committed:
Updated grad_person_extended split.

pyrello committed:
Merge branch 'main' into config_split_2

pyrello committed:
Merge branch 'main' into config_split_2

joewhitsitt committed:
remove core.entity_view_display.node.person.default from complete for grad_person_extended

joewhitsitt committed:
Revert "remove core.entity_view_display.node.person.default from complete for grad_person_extended"

This reverts commit f54dc2f2993b2a94dbeedaf3777a0572aedfb28a.

joewhitsitt committed:
re-export grad_person_extended. retain manual edit from de561414b55f4efb40ac8d4dacc0e74c9f11eb95

Commit:
Export pharmacy site split

Commit:
Update pharmacy site split

Partial split research areas, complete split evd.node.person.default,
remove ef/vd.node.page.default, update palliative views category
to Lists.

joewhitsitt committed:
clean up brand split

joewhitsitt committed:
uno mas por brand webmaster

joewhitsitt committed:
clean up ighn.international site split

joewhitsitt committed:
redo icsa site split. medium_date is custom

Commit:
Remove duplicate create remote_video perm from default

Commit:
Explicitly require config_filter for now, update config_ignore to ^3.0

Commit:
Patch config_ignore for config_filter  cache clear update

Commit:
Uninstall config_filter

Commit:
Remove uiowa_apr.settings from apr split

pyrello committed:
Merged in 'main' branch, resolved conflicts.

pyrello committed:
ighn.setting.yml is no longer a thing, I guess.

Commit:
Remove uiowa_profiles.settings from profiles split

Commit:
Removed ignored config from sitenow_migrate split

Commit:
Partial split evd.node.person.default

Commit:
Make pharmacy_migrate split stackable, export for weight order change

pyrello committed:
Merged 'main' branch, fixed conflicts.

Commit:
Revert "Partial split evd.node.person.default"

This reverts commit 7c62a6f7d8e81aac09c8fae37313e6e2252739d4.

Commit:
Export admissions_migrate split as stackable

Commit:
Export transportation_migrate split as stackable

Commit:
Update and export cs_migrate split as stackable

Commit:
Update and export education_migrate split as stackable

Commit:
Update and export engineering_migrate split as stackable

Commit:
Update and export grad_migrate split as stackable

Commit:
Update and export grad_mdneuroscience_migrate split as stackable

pyrello committed:
Re-exported sitenow_v2 split and manually edited permissions.

Commit:
Update international_migrate split to stackable new format + fix typo in folder and remove

Commit:
Update and export obermann_migrate split as stackable

Commit:
Update and export physics_migrate split as stackable

pyrello committed:
More manual updates to sitenow_v2 permissions.

Commit:
Update and export slis_migrate split as stackable

Commit:
Update and export sppa_migrate split as stackable

pyrello committed:
Re-exported event split.

Commit:
Update and export studentlife_migrate split as stackable

Commit:
Update and export writinguniversity_migrate split as stackable

Commit:
Update and export itu_physics_migrate split as stackable

richardbporter committed:
Merge branch 'main' into config_split_2

Commit:
Shorten migrate config var name

Commit:
Inject migrationPluginManager service and use it to ensure ignored config is created properly

Commit:
Clear plugin definitions on migrate form save, resolves #3871

joewhitsitt committed:
re-export uiowa.edu site split

Commit:
Re-export sppa site split

joewhitsitt committed:
re-export policy.clas site split

joewhitsitt committed:
Merge branch 'config_split_2' of github.com:uiowa/uiowa into config_split_2

Commit:
Remove broken link

Commit:
Minor update on site splits

Commit:
Replace csex Drush alias and use site key in docs

pyrello committed:
Commiting a couple items I'm pretty sure are correct for grad split.

Commit:
STACK AMCS

Commit:
Change education site split weight to 70

Commit:
Stack coronavirus site split, set weight to 70

Commit:
Update config docs with new recommended site split weight

Commit:
Delete education site split

Commit:
Remove core.evd.node.page.default from coronavirus partial split, block uses Site custom category

Commit:
Update coronavirus readme

richardbporter committed:
Merge branch 'main' into config_split_2

pyrello committed:
Added temporary update hook to facilitate easier checking of remaining splits. Slight update to site split definition.

pyrello committed:
Merged 'main' branch, fixed conflicts.

pyrello committed:
Updated admissions split, re-exported, and manually modified some exported files to remove previous errors in copying changes.

pyrello committed:
Re-exported grad site split.

pyrello committed:
Minor changes on export of immuno.grad split.

pyrello committed:
Remove unnecessary override for Admissions.

pyrello committed:
Move person entity_view_display to complete split.

joewhitsitt committed:
update event token display for admissions

joewhitsitt committed:
remove temp update hook

joewhitsitt committed:
clear all the things

pyrello committed:
Updating to using a patch, per best practice.

pyrello committed:
Updated README.
